src/btree/bt_sync.c

```diff
@@ -340,6 +407,13 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 			evict_failed = false;
 			WT_ERR(__wt_reconcile(
 			    session, walk, NULL, WT_REC_CHECKPOINT, NULL));
+
+			/* Track checkpoint progress. */
+			if (++i > 5000 ) {
```

Comment:
replace 5000 with 5 * WT_THOUSAND. Can add a comment explaining why "5000"

Reply:
If we are not tracking checkpoint progress why increment i and set it to 0, ie we can replace the code with:

```
if (track_ckpt_progress && (++i > 5000)) {
     __sync_checkpoint_progress(session);
    i = 0;
}
```

We will also end up doing less comparisons and increments.

Reply:
If track_ckpt_progress is true, then it would lead to two evaluations in the if statements all the time. This if condition is here to periodically check the checkpoint progress. In my opinion there would not be significant gain by changing it.

Reply:
Sulabh was right here - the code is clearer when explicitly checking the variable, and this isn't enough of a performance path to justify removing a comparison.